src/libAtomVM/otp_net.c

```diff
@@ -95,9 +95,41 @@ static term eai_errno_to_term(int err, GlobalContext *glb)
     return term_from_int(err);
 }
 
-static inline term make_address(Context *ctx, struct sockaddr_in *addr)
+static term make_getaddrinfo_result(term *keys, int ai_protocol, int ai_socktype, term inner_addr, GlobalContext *global, Heap *heap)
 {
-    return inet_make_addr4(ntohl(addr->sin_addr.s_addr), &ctx->heap);
+    term result_map;
+    if (term_is_invalid_term(*keys)) {
+        result_map = term_alloc_map(5, heap);
+    } else {
+        result_map = term_alloc_map_maybe_shared(5, *keys, heap);
```

Comment:
I am having a slightly hard time understanding this.  The shared maps, IIRC, are for when you want to create a map that is the same as another map, but they have the same key space (but possibly different values).  Is that what is going on here?  Sorry, I am just a little confused about what the memory optimization is.  Maybe a comment would help?  (Again, I am being dense)

Reply:
Is it better with function comment?

Reply:
Yes that helps.  The shared map is used I guess when the key space is the same.